File: Sesion 8/Ejercicio/Ejercicio2 S8/ejercicio2.java
import java.util.Scanner;

abstract class Reproductor {
    abstract void reproducir();
    abstract void pausar();
    abstract void detener();
}

class ReproductorMP3 extends Reproductor {
    private String dueno;
    private int bateria;

    public ReproductorMP3(String dueno, int bateria) {
        this.dueno = dueno;
        this.bateria = bateria;
    }

    @Override
    void reproducir() {
        System.out.println("Reproduciendo mix del duko 🥵...");
    }

    @Override
    void pausar() {
        System.out.println("Como lo vas a pausar al duko?! Pausando reproducción...");
    }

    @Override
    void detener() {
        System.out.println("No lo saques al dukooooooo!! Deteniendo reproducción...");
    }
}

public class ejercicio2 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Ingrese su nombre: ");
        String nombre = scanner.nextLine();
        System.out.print("Ingrese la batería de su reproductor: ");
        int bateria = scanner.nextInt();
        scanner.nextLine(); // Consumir el salto de línea pendiente

        ReproductorMP3 reproductor1 = new ReproductorMP3(nombre, bateria);

        while (true) {
            System.out.println("Seleccione qué acción desea tener con el Reproductor:");
            System.out.println("1. Reproducir música");
            System.out.println("2. Pausar música");
            System.out.println("3. Detener música");
            System.out.println("4. Salir");
            System.out.print("Elige tu opción: ");
            int opcion = scanner.nextInt();

            switch (opcion) {
                case 1:
                    reproductor1.reproducir();
                    break;
                case 2:
                    reproductor1.pausar();
                    break;
                case 3:
                    reproductor1.detener();
                    break;
                case 4:
                    return;
                default:
                    System.out.println("Opción no válida. Intente de nuevo.");
            }
        }
    }
}

File: Sesion 8/Ejercicio/Ejercicio2 S8/ejercicio2.py
class Reproductor:
    def reproducir(self):
        pass

    def pausar(self):
        pass

    def detener(self):
        pass

class ReproductorMP3(Reproductor):
    def __init__(self, dueno, bateria):
        self.dueno = dueno
        self.bateria = bateria

    def reproducir(self):
        print("Reproduciendo mix del duko 🥵...")

    def pausar(self):
        print("Como lo vas a pausar al duko?! Pausando reproducción...")

    def detener(self):
        print("No lo saques al dukooooooo!! Deteniendo reproducción...")

nombre = input("Ingrese su nombre: ")
bateria = int(input("Ingrese la bateria de su reproductor: "))

reproductor1 = ReproductorMP3(nombre, bateria)

while True:
    print("Seleccione qué acción desea tener con el Reproductor:")
    print("1. Reproducir música")
    print("2. Pausar música")
    print("3. Detener música")
    print("4. Salir")
    opcion = int(input("Elige tu opción: "))

    if opcion == 4:
        break
    elif opcion == 1:
        reproductor1.reproducir()
    elif opcion == 2:
        reproductor1.pausar()
    elif opcion == 3:
        reproductor1.detener()
